src/ape/api/providers.py

```diff
@@ -382,6 +382,23 @@ class ProviderAPI:
     request_header: str
     """A header to set on HTTP/RPC requests."""
 
+    def __post_init__(self):
```

Comment:
Sooo we will have to update `infura` and `alchemy` to call `super().__post_init__()` to get this to fully work.

Altso, `mypy` still don't know about these newly added members, I am not sure if there is a hack for that but I am cool with `# type: ignore` comments because _we_ know these methods exist.

Reply:
If we like this approach, I will get the PRs ready for those 2 plugins so we remember to fix them after we release

Reply:
Another nice thing is that pydantic update will fix some of these issues